docs/articles/vector_search_based_recommender_system.md

```diff
@@ -0,0 +1,276 @@
+### Backend Overview
+
+Flask makes creating web applications in Python easier by rendering the HTML and CSS files via single-line commands. You can check the backend code that Flask utilizes [here](https://github.com/ankit1khare/rockset-vector-search/blob/main/app.py).
+
+Initially, the Get method will be called and the HTML file will be rendered. As there will be no recommendation at this time, the basic structure of the page will be displayed on the browser. After this is executed, we can fill the form and submit it thereby utilizing the POST method to get some recommendations.
+
+Let's dive into the main components of the code as we did for the front-end:
+
+1.  **Flask App Setup:**
+    1.  A Flask application named app is defined along with a route for both GET and POST requests at the root URL ("/").
+
+2.  **Index function:**
+    1.  Function built to primarily handle both GET and POST requests.
+    2.  If it's a POST request:
+        1.  Extracts form data from the frontend.
+        2.  Calls a set of functions to process the input data and fetch recommended results from Rockset database.
+        3.  Fetches recommended product images by searching through all the product images saved in this [directory](https://github.com/ankit1khare/rockset-vector-search/tree/main/static).
+        4.  Renders the index.html template with the results.
+    3.  If it's a GET request:
+        1.  Renders the index.html template with the search form.
+```python
+
+@app.route('/', methods=['GET', 'POST'])
+def index():
+    if request.method == 'POST':
+        # Extract data from form fields
+        inputs = get_inputs()
+
+        search_query_embedding = get_openai_embedding(inputs, client)
+        rockset_key = os.environ.get('ROCKSET_API_KEY')
+        region = Regions.usw2a1
+        records_list = get_rs_results(inputs, region, rockset_key, search_query_embedding)
+
+        folder_path = 'static'
+        for record in records_list:
+            # Extract the identifier from the URL
+            identifier = record["image_url"].split('/')[-1].split('_')[0]
+            file_found = None
+            for file in os.listdir(folder_path):
+                if file.startswith(identifier):
+                    file_found = file
+                    break
+            if file_found:
+                # Overwrite the record["image_url"] with the path to the local file
+                record["image_url"] = file_found
+                record["description"] = json.dumps(record["description"])
+                # print(f"Matched file: {file_found}")
+            else:
+                print("No matching file found.")
+
+        # Render index.html with results
+        return render_template('index.html', records_list=records_list, request=request)
+
+    # If method is GET, just render the form
+    return render_template('index.html', request=request)
+```
+
+3. **Data Processing Functions:**
+    1. get_inputs(): Extracts form data from the request.
```

Comment:
why are we back to 1. here? Should this be 3.?

Reply:
@fatirmalik1
why are we back to 1. here? Should this be 3.?

Reply:
@robertdhayanturner , the intent is to have these as sub-bullets.